sqlmesh/core/state_sync/base.py

```diff
@@ -14,9 +16,28 @@
     SnapshotNameVersionLike,
     SnapshotTableInfo,
 )
+from sqlmesh.utils import optional_import
 from sqlmesh.utils.date import TimeLike, now, to_datetime
 from sqlmesh.utils.errors import SQLMeshError
 
+MIGRATIONS = []
+MIGRATION_VERSION = 0
+
+while True:
+    migration = optional_import(f"sqlmesh.migrations._{MIGRATION_VERSION:04}")
```

Comment:
This looks weird. Is there a better way to list all modules in a package and then just get the size of the list instead of doing increments?

Reply:
1. it needs to be sorted
2. didn't find an easy way to do that. you have to walk the actual path

Reply:
> you have to walk the actual path

this doesn't sound so bad